src/encoded/submit.py

```diff
@@ -760,18 +783,252 @@ def create_json_out(self):
+                                    phenotypes.extend(member_phenotypes)
+                                    phenotypes = list(set(phenotypes))
+                                    if len(phenotypes) >= 4:
+                                        break
+                        if phenotypes:
+                            final_family_dict[match['@id']]['family_phenotypic_features'] = phenotypes[:4]
+                        del final_family_dict[match['@id']]['proband']
+                        del final_family_dict[match['@id']]['aliases']
+        return final_family_dict
+
+    def check_individuals(self):
+        """Make sure that every value in mother ID or father ID columns are also in sheet in same family.
+        If a mother or father ID does not have a line in the sheet, just create minimal metadata for it
+        and add it to the family."""
+        parent_dict = {'mother': 'F', 'father': 'M'}
+        for fam_alias, fam_metadata in self.families.items():
+            for member in fam_metadata['members']:
+                individual = self.individuals[member]
+                for parent in ['mother', 'father']:
+                    if individual.get(parent):
+                        if individual[parent] not in fam_metadata['members']:
+                            info = {
+                                'individual_id': individual[parent],
+                                'sex': parent_dict[parent]
+                            }
+                            self.individuals[individual[parent]] = info
+                            fam_metadata['members'].append(individual[parent])
+
+    def process_rows(self):
+        """
+        Method for iterating over spreadsheet rows to process each one and compare it to previous rows.
+        """
+        for i, row in enumerate(self.rows):
+            try:
+                processed_row = PedigreeRow(row, i + 1 + self.counter, self.project, self.institution)
+                self.errors.extend(processed_row.errors)
+                self.add_individual_metadata(processed_row.individual)
+            except AttributeError as e:
+                continue
+        self.families = self.add_family_metadata()
+        self.check_individuals()
+
+    def create_json_out(self):
+        """
+        Creates final json that can be used for subsequent validation function.
+        """
+        itemtype_dict = {
+            'family': self.families,
+            'individual': self.individuals
+        }
+        for key in itemtype_dict:
+            self.json_out[key] = {}
+            for alias, metadata in itemtype_dict[key].items():
+                new_metadata = {k: v for k, v in metadata.items() if v}
+                new_metadata['project'] = self.project_atid
+                new_metadata['institution'] = self.institution_atid
+                if key == 'individual' and 'proband' in new_metadata:
+                    del new_metadata['proband']
```

Comment:
Purely cosmetic, but you _could_ alternatively write this as (untested):
```suggestion
                if key == 'individual':
                    new_metadata.pop('proband', None)
```
The reason I note it is that it only has to make one pass through the dictionary to do it this way, as opposed to one pass for the `in` and one pass for the `del`. It's not a huge detail and I would mostly pick this style because it's a bit more visually compact.

The `None` argument is required in the `pop` because otherwise it will err if not present. Unlike `del`, `.pop` can be used for value and it's the default value, which doesn't matter since this usage is not actually being used for value. It only matters that you pass _something_ in that arg because the 1-arg and 2-arg functionalities of `.pop` are different.